wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/provisioncredential/ProvisionCredentialScreen.kt

```diff
@@ -149,12 +148,26 @@ fun ProvisionDocumentScreen(
                     }
 
                     is EvidenceRequestMessage -> {
-                        EvidenceRequestMessageView(
-                            evidenceRequest,
-                            provisioningViewModel = provisioningViewModel,
-                            walletServerProvider = walletServerProvider,
-                            documentStore = documentStore
-                        )
+                        // show a standard message that does not have a [messageType] defined
+                        if (evidenceRequest.messageType==null) {
```

Comment:
I'm wondering if this would be simpler and more maintainable with a new EvidenceRequest type instead of switching based on messageType. Something like EvidenceRequestCompleted or EvidenceRequestCompletedMessage.

The current implementation adds messageType and messageTitle to EvidenceRequestMessage when the existing messages wouldn't use it. And it uses a different Composable based on one of those parameters (EvidenceRequestCompleteScreen or EvidenceRequestMessageView), which in all other cases here is mapped to the EvidenceRequest class type. That adds some unexpected complexity to this code here.

If, by contrast, we had a separate EvidenceRequest type (say, EvidenceRequestCompletedMessage), then we'd only use messageTitle where it was needed (so nobody would wonder why nothing was changing for a EvidenceRequestMessageView when they change the title) and the mapping here in ProvisionDocumentScreen would be similar to all the other mappings, so there's less to think about for anyone who needs to come in and maintain the code.

Reply:
as per our discussion, I've reverted changes to the existing `EvidenceRequestMessage` and created a new type of `EvidenceRequest` subclass for detailed messages (having a message classification, etc..) that allows showing the "completed" screen at the end as well as any other type of detailed message without having to create yet another subclass of `EvidenceRequest`